core/hoverfly_funcs.go

```diff
@@ -117,6 +114,65 @@ func (hf *Hoverfly) GetResponse(requestDetails models.RequestDetails) (*models.R
 	return &response, nil
 }
 
+func (hf *Hoverfly) applyBodyTemplating(requestDetails *models.RequestDetails, response *models.ResponseDetails, cachedResponse *models.CachedResponse) (string, error) {
+	var template *raymond.Template
+	if cachedResponse != nil && cachedResponse.ResponseTemplate != nil {
+		template = cachedResponse.ResponseTemplate
+	} else {
+		// Parse and cache the template
+		template, _ = hf.templator.ParseTemplate(response.Body)
+		if cachedResponse != nil {
+			cachedResponse.ResponseTemplate = template
+		}
+	}
+
+	return hf.templator.RenderTemplate(template, requestDetails, hf.state.State)
+}
+
+func (hf *Hoverfly) applyHeadersTemplating(requestDetails *models.RequestDetails, response *models.ResponseDetails, cachedResponse *models.CachedResponse) (map[string][]string, error) {
+	var headersTemplates map[string][]*raymond.Template
+	if cachedResponse != nil && cachedResponse.ResponseHeadersTemplates != nil {
+		headersTemplates = cachedResponse.ResponseHeadersTemplates
+	} else {
+		var header []*raymond.Template
+		headersTemplates = map[string][]*raymond.Template{}
+		// Parse and cache headers templates
+		for k, v := range response.Headers {
+			header = make([]*raymond.Template, len(v))
+			for i, h := range v {
+				header[i], _ = hf.templator.ParseTemplate(h)
+			}
+
+			headersTemplates[k] = header
+		}
+
+		if cachedResponse != nil {
+			cachedResponse.ResponseHeadersTemplates = headersTemplates
+		}
+	}
+
+	var (
+		header []string
+		err error
+	)
+	headers := map[string][]string{}
+
+	// Render headers templates
+	for k, v := range headersTemplates {
+		header = make([]string, len(v))
```

Comment:
I meant you can do `header := make([]string, len(v))` here, it will create a slice with variable length, and you don't need the `var` declaration in line 155.  Anyway, it's not a big deal.